tests/end_to_end/test_kakarot.py

```diff
@@ -156,321 +135,231 @@ async def test_execute(
-            except Exception as e:
-                print(e)
-            new_class_hash = await starknet.get_class_hash_at(kakarot.address)
-            assert prev_class_hash == new_class_hash
-
-        async def test_should_raise_when_class_hash_is_not_declared(
-            self, starknet, kakarot, invoke
-        ):
-            prev_class_hash = await starknet.get_class_hash_at(kakarot.address)
-            await invoke("kakarot", "upgrade", 0xDEAD)
-            new_class_hash = await starknet.get_class_hash_at(kakarot.address)
-            assert prev_class_hash == new_class_hash
+        async def test_should_raise_when_caller_is_not_owner(self, other, class_hashes):
+            tx_hash = await invoke(
+                "kakarot", "upgrade", class_hashes["EVM"], account=other
+            )
+            receipt = await RPC_CLIENT.get_transaction_receipt(tx_hash)
+            assert receipt.execution_status.name == "REVERTED"
+            assert "Ownable: caller is not the owner" in receipt.revert_reason
 
-        async def test_should_upgrade_class_hash(
-            self, starknet, kakarot, invoke, class_hashes
-        ):
-            prev_class_hash = await starknet.get_class_hash_at(kakarot.address)
+        async def test_should_raise_when_class_hash_is_not_declared(self):
+            tx_hash = await invoke("kakarot", "upgrade", 0xDEAD)
+            receipt = await RPC_CLIENT.get_transaction_receipt(tx_hash)
+            assert receipt.execution_status.name == "REVERTED"
+            assert "is not declared" in receipt.revert_reason
+
+        async def test_should_upgrade_class_hash(self, kakarot, class_hashes):
+            prev_class_hash = await RPC_CLIENT.get_class_hash_at(kakarot.address)
             await invoke("kakarot", "upgrade", class_hashes["replace_class"])
-            new_class_hash = await starknet.get_class_hash_at(kakarot.address)
+            new_class_hash = await RPC_CLIENT.get_class_hash_at(kakarot.address)
             assert prev_class_hash != new_class_hash
             assert new_class_hash == class_hashes["replace_class"]
             await invoke("kakarot", "upgrade", prev_class_hash)
 
     class TestTransferOwnership:
-        @SyscallHandler.patch("Ownable_owner", 0xDEAD)
-        async def test_should_raise_when_caller_is_not_owner(
-            self, kakarot, invoke, other
-        ):
-            prev_owner = await kakarot.functions["get_owner"].call()
-            try:
-                await invoke("kakarot", "transfer_ownership", account=other)
-            except Exception as e:
-                print(e)
-            new_owner = await kakarot.functions["get_owner"].call()
+        async def test_should_raise_when_caller_is_not_owner(self, kakarot, other):
+            prev_owner = (await kakarot.functions["get_owner"].call()).owner
+            await invoke("kakarot", "transfer_ownership", other.address, account=other)
```

Comment:
Not sure to understand. 
Why the explicit patch for the owner was removed ?
This test should raise as the caller should not be the current owner of kakarot.
Am I missing something ?

Reply:
this is an end to end test, patch have no effect here